test/Vault.spec.ts

```diff
@@ -501,43 +501,75 @@ describe("Vault", () => {
 
     it("reverts if no investable amount", async () => {
       await vault.connect(owner).setStrategy(strategy.address);
-      await addYieldToVault("10");
-      await underlying.mint(strategy.address, parseUnits("100"));
 
       await expect(vault.connect(owner).updateInvested()).to.be.revertedWith(
         "Vault: nothing to invest"
       );
     });
 
-    it("moves the funds to the strategy", async () => {
-      await vault.connect(owner).setStrategy(strategy.address);
-      await vault.connect(owner).setInvestPct("8000");
-      await addYieldToVault("100");
+    describe("invest scenario", () => {
+      it("moves the funds to the strategy", async () => {
+        await vault.connect(owner).setStrategy(strategy.address);
+        await vault.connect(owner).setInvestPct("8000");
+        await addYieldToVault("100");
 
-      await vault.connect(owner).updateInvested();
+        await vault.connect(owner).updateInvested();
 
-      expect(await underlying.balanceOf(strategy.address)).to.eq(
-        parseUnits("80")
-      );
+        expect(await underlying.balanceOf(strategy.address)).to.eq(
+          parseUnits("80")
+        );
+      });
+
+      it("emits an event", async () => {
+        await vault.connect(owner).setStrategy(strategy.address);
+        await vault.connect(owner).setInvestPct("8000");
+        await addYieldToVault("100");
+
+        const tx = await vault.connect(owner).updateInvested();
+
+        await expect(tx).to.emit(vault, "Invested").withArgs(parseUnits("80"));
+      });
     });
 
-    it("emits an event", async () => {
-      await vault.connect(owner).setStrategy(strategy.address);
-      await vault.connect(owner).setInvestPct("8000");
-      await addYieldToVault("100");
+    describe("disinvest scenario", () => {
+      it("call strategy.withdrawToVault with required amount", async () => {
+        await vault.connect(owner).setStrategy(strategy.address);
+        await addYieldToVault("10");
+        await underlying.mint(strategy.address, parseUnits("190"));
 
-      const tx = await vault.connect(owner).updateInvested();
+        await vault.connect(owner).updateInvested();
 
-      await expect(tx).to.emit(vault, "Invested").withArgs(parseUnits("80"));
+        expect(await underlying.balanceOf(vault.address)).to.eq(parseUnits("20"));
+        expect(await underlying.balanceOf(strategy.address)).to.eq(
+          parseUnits("180")
+        );
+      });
+
+      it("emits an event", async () => {
+        await vault.connect(owner).setStrategy(strategy.address);
+        await addYieldToVault("10");
+        await underlying.mint(strategy.address, parseUnits("190"));
+
+        const tx = await vault.connect(owner).updateInvested();
+
+        await expect(tx).to.emit(vault, "Disinvested").withArgs(parseUnits("10"));
+      });
     });
   });
 
+  // TODO: change
   describe("investableAmount", () => {
```

Comment:
The function name is not `investableAmount` anymore. Is it ok to have a testing block with this name? Or might be confusing? Should we merge this tests in the next testing block?

Reply:
good shout overlooked this :see_no_evil:

Reply:
merged it in with the `investState` test case describe block since they're testing the same core functionality